### ISSUE TITLE
Make backend more scalable

### ISSUE THREAD

**Issue by gbougakov** [gbougakov/slack-yellbot#4]
Right now, when user yells a message, Yell Bot creates a database entry that looks like this for usergroups:
```json
{
    "message": { ... },
    "intendedUserGroup": "GROUPID",
    "readBy": [ ... ],
    ...
}
```
and for channels:
```json
{
    "message": { ... },
    "intendedChannel": "CHANNELID",
    "readBy": [ ... ],
    ...
}
```

After this issue is closed, when user yells a message, Yell Bot fetches a user list for the specified usergroup/channel/etc and creates a database entry that looks like this: 
```json
{
    "message": { ... },
    "mustBeReadBy": [ ... ],
    "readBy": [ ... ],
    ...
}
```

This update will:
1. Minimize the amount of API calls to Slack
2. Consolidate the code that reminds users, because right now it's separate for usergroups and channels
3. Make Yell Bot more scalable by making it easier to add new features without polluting code

**Comment by gbougakov** [comment]
Code is in `consolidate` branch

**Comment by gbougakov** [comment]
Closed with #5
